gui/src/components/onboarding/pages/trackers-assign/TrackerAssignOptions.tsx

```diff
@@ -0,0 +1,125 @@
+import { Radio } from '@/components/commons/Radio';
+import { Typography } from '@/components/commons/Typography';
+import { AssignMode, defaultConfig, useConfig } from '@/hooks/config';
+import { ASSIGNMENT_MODES } from '@/components/onboarding/BodyAssignment';
+import { useLocalization } from '@fluent/react';
+import { useTrackers } from '@/hooks/tracker';
+import { useForm } from 'react-hook-form';
+import { useEffect } from 'react';
+import { Dropdown } from '@/components/commons/Dropdown';
+
+// Ordered collection of assign modes with the number of IMU trackers
+const ASSIGN_MODE_OPTIONS = [
+  AssignMode.LowerBody,
+  AssignMode.Core,
+  AssignMode.EnhancedCore,
+  AssignMode.FullBody,
+  AssignMode.All,
+].reduce(
+  (options, mode) => ({ ...options, [mode]: ASSIGNMENT_MODES[mode].length }),
+  {} as Record<AssignMode, number>
+);
+
+export function TrackerAssignOptions({
+  variant = 'radio',
+}: {
+  variant: 'radio' | 'dropdown';
+}) {
+  const { l10n } = useLocalization();
+  const { useConnectedIMUTrackers } = useTrackers();
+  const connectedIMUTrackers = useConnectedIMUTrackers().length;
+
+  const { config, setConfig } = useConfig();
+  const { control, watch, setValue } = useForm<{
+    assignMode: AssignMode;
+  }>({
+    defaultValues: {
+      assignMode: config?.assignMode ?? defaultConfig.assignMode,
+    },
+  });
+  const { assignMode } = watch();
+
+  useEffect(() => {
+    setConfig({ assignMode });
+  }, [assignMode]);
+
+  useEffect(() => {
+    if (connectedIMUTrackers <= ASSIGN_MODE_OPTIONS[assignMode]) return;
+
+    const selectedAssignMode =
+      (Object.entries(ASSIGN_MODE_OPTIONS).find(
+        ([_, count]) => count >= connectedIMUTrackers
+      )?.[0] as AssignMode) ?? AssignMode.All;
+
+    if (assignMode !== selectedAssignMode) {
+      setValue('assignMode', selectedAssignMode);
+    }
+  }, [connectedIMUTrackers, assignMode]);
+
+  const ItemContent = ({
+    mode,
+    trackersCount,
+  }: {
+    mode: string;
+    trackersCount: number;
+  }) => (
+    <>
+      <Typography variant="mobile-title" textAlign="text-right">
```

Comment:
I'll suggest to switch variant from mobile to "main-title" to keep big numbers. As on mobile we will have drop down, you dopn't need to save space ;)